### PR TITLE
Run function check and out check in TestTensorDeviceOps

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -19259,6 +19259,8 @@ def inner(self, device, dtype):
 # - cpu_dtype_list (=[]), a list of torch dtypes to test the op(s) on cpu
 # - make_inplace_variant (=True), if true the inplace version of the op (op_) is also tested
 # - decorators (=[]), a list of decorators to apply to the test
+# - self_position (=-1), the position of self in the arg list, -1 means skip function check
+# - test_out (=False), whether to test the out= version of the operator
```
Comment (zasdfgbnm) [inline test/test_torch.py:19263]: Currently only support addmm and addmv, more are coming in future PR

Comment (mruberry) [reply test/test_torch.py:19263]: We're planning on killing this test suite soonish and refactoring the existing tests using OpInfos (you can see test_unary_ufunc already has copies of al the previous tests) so I wouldn't plan further improvements here.

Comment (mruberry) [reply test/test_torch.py:19263]: (This PR, however, seems fine.)

File: test/test_torch.py [hunk 2]
```diff
@@ -19658,6 +19662,25 @@ def fn(self, device, dtype) -> None:
...
+        if self_position >= 0:
+            cpu_args.insert(self_position, cpu_tensor)
+            device_args.insert(self_position, device_tensor)
+            cpu_function_result = getattr(torch, op_str)(*cpu_args)
+            device_function_result = getattr(torch, op_str)(*device_args)
+            self.assertEqual(cpu_result, cpu_function_result, atol=precision, rtol=0)
+            self.assertEqual(device_result, device_function_result, atol=precision, rtol=0)
+
+        # check method matches with function(out)
+        if test_out:
```
Comment (mruberry) [inline test/test_torch.py:19675]: This only works if the preceding check is true, right? Should it be nested under it?

Comment (zasdfgbnm) [reply test/test_torch.py]: Yes, you are right. Fixed.

**Review by mruberry — APPROVED** [review]
Cool!

**Pull request closed without merging**